shared/src/store/action-reducers/transfer.ts

```diff
@@ -0,0 +1,146 @@
+    public readonly elementId!: UUID;
+
+    @IsUUID(4, uuidValidationOptions)
+    @IsOptional()
+    public readonly targetTransferPointId?: UUID;
+
+    @IsInt()
+    @IsOptional()
+    /**
+     * How much time in ms should be added to the transfer time.
+     * If it is negative, the transfer time will be decreased.
+     * If the time set the the end of the transfer time to the past, it sets it to the current time.
+     */
+    public readonly timeToAdd?: number;
+}
+
+export class TogglePauseTransferAction implements Action {
+    @IsString()
+    public readonly type = '[Transfer] Toggle pause transfer';
+
+    @IsString()
+    elementType!: 'personnel' | 'vehicles';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly elementId!: UUID;
+}
+
+export namespace TransferActionReducers {
+    export const addToTransfer: ActionReducer<AddToTransferAction> = {
+        action: AddToTransferAction,
+        reducer: (
+            draftState,
+            {
+                elementType,
+                elementId,
+                startTransferPointId,
+                targetTransferPointId,
+            }
+        ) => {
+            const element = getElement(draftState, elementType, elementId);
+            if (element.transfer) {
+                throw new ReducerError(
+                    `Element with id ${element.id} is already in transfer`
+                );
+            }
+            const startTransferPoint = getElement(
+                draftState,
+                'transferPoints',
+                startTransferPointId
+            );
+            const connection =
+                startTransferPoint.reachableTransferPoints[
+                    targetTransferPointId
+                ];
+            if (!connection) {
+                throw new ReducerError(
+                    `TransferPoint with id ${targetTransferPointId} is not reachable from ${startTransferPointId}`
+                );
+            }
+            // The element is now in transfer
+            delete element.position;
+            element.transfer = {
+                startTransferPointId,
+                targetTransferPointId,
+                endTimeStamp: draftState.currentTime + connection.duration,
+                isPaused: false,
+            };
+            return draftState;
+        },
+        rights: 'participant',
+    };
+
+    export const editTransfer: ActionReducer<EditTransferAction> = {
+        action: EditTransferAction,
+        reducer: (
+            draftState,
+            { elementType, elementId, targetTransferPointId, timeToAdd }
+        ) => {
+            const element = getElement(draftState, elementType, elementId);
+            if (!element.transfer) {
+                throw getNotInTransferError(element.id);
+            }
+            if (targetTransferPointId) {
+                element.transfer.targetTransferPointId = targetTransferPointId;
+            }
+            if (timeToAdd !== undefined) {
```

Comment:
Why not

```suggestion
            if (timeToAdd) {
```

as above? Having `timeToAdd === 0`, which would then evaluate the above to `false` would be fine as adding `0` time has no effect, so we don't have to calculate a new time.